### ISSUE TITLE
Objective C-Code on Darwin breaks GCC

### ISSUE THREAD

**Issue by ax3l** [libuv/libuv#2805]
The file `src/unix/darwin-proctitle.c` includes the macOS system framework headers
```cxx
# include <CoreFoundation/CoreFoundation.h>
# include <ApplicationServices/ApplicationServices.h>
```
which are written at least partly on Objective-C.

Due to that, GCC builds on macOS fail with errors such as:
```cxx
/Library/Developer/CommandLineTools/SDKs/MacOSX10.14.sdk/System/Library/Frameworks/Security.framework/Headers/Authorization.h:193:7: error: variably modified 'bytes' at file scope
    char bytes[kAuthorizationExternalFormLength];
```

Full details:
[output.txt](https://github.com/libuv/libuv/files/4519461/spack-build-out.txt)
[environment.txt](https://github.com/libuv/libuv/files/4519460/spack-build-env.txt)

The problem is essentially that some syntax that the pre-processor includes is not valid C and somehow AppleClang is optimistically identifying this correctly as Objective-C translation unit and GCC is not (without further help).

Maybe renaming the file `src/unix/darwin-proctitle.c` -> `src/unix/darwin-proctitle.m` could already help?

* **Version**: any recent
* **Platform**: seen on macOS 10.14 and 10.15 with any GCC

Cross-Refs.:

* https://gitlab.kitware.com/cmake/cmake/-/issues/20620
* https://github.com/spack/spack/issues/12644

**Comment by ax3l** [comment]
Fix proposed in #2808

**Comment by bnoordhuis** [comment]
Requiring an objective-c compiler is not on the table but I'm open to two alternatives:

1. Skip compiling darwin-proctitle.c when the compiler isn't capable of handling SDK headers.

2. Removing the `<CoreFoundation/CoreFoundation.h>` and `<ApplicationServices/ApplicationServices.h>` by defining the types inline.

I'll open a PR for the second alternative.
